### ISSUE TITLE
Store gitops runtime definition in git

### ISSUE THREAD

**Issue by markramm** [weaveworks/weave-gitops#101]
**Short description**
As a platform operator I want to check the platform configuration into a repository (to start the same one that contains the apps)  so that configuration is continuously reconciled by the gitops engine, and can be updated from git. (Preventing operational errors, and enabling faster recovery when errors are introduced). '

**Gitops Platform Repo:**  The repository where the gitops runtime is stored. 
**Gitops Application Repo:**  The repository where apps are added.   Apps either be local to this repo, or reference another repository which contains that apps config. 
**Gitops Automations:**  The source, helm, and customization controller settings that automates the gitops runtime for an set of declarative models stored in git (Apps and gitops runtime).   

The default setup should be a monorepo which contains both the app repo and the platform repo.  

If you don't specify a URL we are not going to store the Gitops Runtime config (gort). 

CLI experience: https://miro.com/app/board/o9J_l7vcDC8=/

**User Story**

As a the operator responsible for setting up gitops (platform operator) I would like to run `wego gitops install` and supply it with an empty repository and get fully functioning weave gitops runtime installed in the cluster, with application config AND platform config stored in that repository. 

There are two additional cases which need to be considered:   

- The platform may be managed by a separate team than the applications, in which case it may be desirable to split the Wego Platform and Wego App definitions into separate repositories.  
- Wego may already be installed in a cluster and we may need to retrieve the gitops runtime config from the cluster, persist it to git, and update the cluster.  This shoud not be implemented in this ticket, it will be split out into another ticket. 

I think there are four things that need to happen, some of which can be automated together: 

* get platform definition from the running cluster
* put that definition in the right place in the repo
* create the the platform sync definition
* put that definition in the repo, and update flux to use definition from the repo 

**Acceptance criteria**
- [ ] AC/1
**Given** that I have a wego binary with a working install command
**And** I have a `.kube/config` file that references an active Kubernetes cluster
**And** I have a platform repository or an app repository with infraRepo=true in its config file
**When** I run`wego install` against the active cluster
**Then** the gitops runtime manifests (CRDs, CRs) are added to the cluster and stored in the repository
**And** any changes made to the manifests will be applied to the cluster via flux when they are merged
- [ ] AC/2
**Given** that I have a wego binary with a working install command
**And** I have a `.kube/config` file that references an active Kubernetes cluster
**And** I **do not** have a platform repository or an app repository with infraRepo=true in its config file
**And** GITHUB_ORG is set in the environment
**When** I run`wego install` against the active cluster without parameters
**Then** a private repository is created with a name derived from the active cluster, in the org referenced by GITHUB_ORG
**And** the gitops runtime manifests (CRDs, CRs) are added to the cluster and stored in the repository
**And** any changes made to the manifests will be applied to the cluster via flux when they are merged
**When** I run`wego install` against the active cluster with any of `--organization`, `--public`, `--repository-name` 
**Then** a repository is created, with default values overridden by the arguments
**And** the gitops runtime manifests (CRDs, CRs) are added to the cluster and stored in the repository
**And** any changes made to the manifests will be applied to the cluster via flux when they are merged
- [ ] All the configuration needed to re-setup weave gitops on the cluster is stored in git 
- [ ] Any drift from upstream gitops platform definitions are either "reconciled away" or are reported as errors. 

**Design materials**
- Screenshots
- Design diagrams
- WEP: https://github.com/weaveworks/weave-gitops-private/blob/wep-wego-recover/docs/weps/WEP-002-Weave-gitops-recover.md 
- PR/FAQ

**Work breakdown**
(optional)Task breakdown of the work
- [ ] _create X_
- [ ] _design X_
- [ ] _..._

**Comment by palemtnrider** [comment]
@jrryjcksn add the given-when-then syntax so we can refine.

**Comment by markramm** [comment]
We will try to break this up into two smaller stories

**Comment by markramm** [comment]
Need some UX discussion on how we identify the platform repository. 

We are going to expose the .wego platform repository up front.   

Platform folder should be configurable.

**Comment by jpellizzari** [comment]
Depends on #560
